src/action/common/create_users_and_group.rs

```diff
@@ -106,57 +117,83 @@ impl Action for CreateUsersAndGroup {
         create_group.try_execute().await?;
 
         // Mac is apparently not threadsafe here...
-        for create_user in create_users.iter_mut() {
-            // let mut create_user_clone = create_user.clone();
-            // let _abort_handle = set.spawn(async move {
-            create_user.try_execute().await?;
-            //     Result::<_, CreateUserError>::Ok((idx, create_user_clone))
-            // });
-        }
+        use target_lexicon::OperatingSystem;
+        match OperatingSystem::host() {
+            OperatingSystem::MacOSX {
+                major: _,
+                minor: _,
+                patch: _,
+            }
+            | OperatingSystem::Darwin => {
+                for create_user in create_users.iter_mut() {
+                    create_user.try_execute().await?;
+                }
+            },
+            _ => {
+                let mut set = JoinSet::new();
+                let mut errors: Vec<Box<dyn std::error::Error + Send + Sync>> = Vec::new();
+                for (idx, create_user) in create_users.iter_mut().enumerate() {
+                    let mut create_user_clone = create_user.clone();
+                    let _abort_handle = set.spawn(async move {
+                        create_user_clone.try_execute().await?;
+                        Result::<_, _>::Ok((idx, create_user_clone))
+                    });
+                }
+
+                while let Some(result) = set.join_next().await {
+                    match result {
+                        Ok(Ok((idx, success))) => create_users[idx] = success,
+                        Ok(Err(e)) => errors.push(e),
+                        Err(e) => return Err(e)?,
+                    };
+                }
 
-        // while let Some(result) = set.join_next().await {
-        //     match result {
-        //         Ok(Ok((idx, success))) => create_users[idx] = success,
-        //         Ok(Err(e)) => errors.push(e),
-        //         Err(e) => return Err(e)?,
-        //     };
-        // }
-
-        // if !errors.is_empty() {
-        //     if errors.len() == 1 {
-        //         return Err(errors.into_iter().next().unwrap().into());
-        //     } else {
-        //         return Err(CreateUsersAndGroupError::CreateUsers(errors));
-        //     }
-        // }
+                if !errors.is_empty() {
+                    if errors.len() == 1 {
+                        return Err(errors.into_iter().next().unwrap().into());
```

Comment:
This doesn't immediately make sense to me -- why do we wrap one with our own error type and not this one?

Reply:
Ahh this is because returning multiple errors like this provides the user a degraded viewing experience. When we have multiple errors we can't provide as rich of an error chain for eyre, and we kind of just blah everywhere. It's less desirable, so where possible we should still provide a quality experience.